typedoc.json

```diff
@@ -0,0 +1,10 @@
+{
+  "$schema": "https://typedoc.org/schema.json",
+  "out": "./docs/reference",
+  "entryPoints": [
+    "./src/index.ts"
+  ],
+  "plugin": ["typedoc-plugin-markdown"],
+  "gitRevision": "main",
+  "publicPath": "/reference/"
```

Comment:
Since I stuck the docs into docs/reference adding this makes the paths still work in vitepress.

I didn't add it to vitepress sidebar or anything since I figured the docs team later might want to completely remove the docs generation. This is just giving them a head start and also a stop gap until they can help